sklearnex/decomposition/pca.py

```diff
@@ -14,4 +14,348 @@
 # limitations under the License.
 # ===============================================================================
 
-from daal4py.sklearn.decomposition import PCA
+import logging
+
+from daal4py.sklearn._utils import daal_check_version
+
+if daal_check_version((2024, "P", 100)):
+    import numbers
+    from math import sqrt
+
+    import numpy as np
+    from scipy.sparse import issparse
+    from sklearn.utils.validation import check_is_fitted
+
+    from daal4py.sklearn._n_jobs_support import control_n_jobs
+    from daal4py.sklearn._utils import sklearn_check_version
+
+    from .._device_offload import dispatch, wrap_output_data
+    from .._utils import PatchingConditionsChain
+
+    if sklearn_check_version("1.1") and not sklearn_check_version("1.2"):
+        from sklearn.utils import check_scalar
+
+    from sklearn.decomposition import PCA as sklearn_PCA
+
+    from onedal.decomposition import PCA as onedal_PCA
+
+    @control_n_jobs(decorated_methods=["fit", "transform"])
+    class PCA(sklearn_PCA):
+        __doc__ = sklearn_PCA.__doc__
+
+        if sklearn_check_version("1.2"):
+            _parameter_constraints: dict = {**sklearn_PCA._parameter_constraints}
+
+        if sklearn_check_version("1.1"):

```

Comment:
What about deriving from different base according to the version?

Reply:
Good suggestion, will be implemented as part of a later PR